### PR TITLE
[XHarness] Do not build the test assemblies for the iOS platforms.

### PR DESCRIPTION
**Pull request by mandel-macaque**
We now download the ios Mono sdk which means that there is no need for
us to build the test assemblies on iOS, WatchOS and tvOS BUT we need to
do so for the Mac tests.

The change includes:
  1. Use fo the downloaded test assemblies.
  2. Removal of the dependency of building them in the tests.
  3. Move back to using reflection for the project generation.
  4. Remove the cached project details added in the workaround.

PS: Assemblies use lower case in the SDK and some tests had to be updated.

### CODE REVIEW

**Comment by mandel-macaque** [comment]
@rolfbjarne AFAIK we are not yet downloading the Mac SDK, correct?

File: tests/xharness/Harness.cs [hunk 1]
```diff
@@ -34,6 +34,11 @@ public class Harness
 			get { return Path.GetFullPath (Path.Combine (RootDirectory, "..", "tools", "xibuild", "xibuild")); }
 		}
 
+		public string DownloadsPath
+		{
+			get { return Path.GetFullPath (Path.Combine (RootDirectory, "..", "builds", "downloads")); }
```
Comment (rolfbjarne) [inline tests/xharness/Harness.cs]: You can get the location of the downloaded SDK like this: https://gist.github.com/rolfbjarne/f5108a574f46dd2fcbe639d84c02e3d1

That should simplify some code (and it'll work even if you end up with multiple downloads after mono is bumped).

File: tools/bcl-test-importer/BCLTestImporter/BCLTestProjectGenerator.cs [hunk 2]
```diff
@@ -260,6 +258,24 @@ public BCLTestProjectGenerator (string outputDirectory, string monoRootPath, str
 			RegisterTypesTemplatePath = registerTypesTemplatePath ?? throw new ArgumentNullException (nameof (registerTypesTemplatePath));
 		}
 
+		string GetReleaseDownload (Platform platform)
+		{
+			if (string.IsNullOrEmpty (DownloadsPath))
+				return null;
+			switch(platform) {
```
Comment (rolfbjarne) [inline tools/bcl-test-importer/BCLTestImporter/BCLTestProjectGenerator.cs]: Style: space before opening parenthesis.

File: tests/xharness/Harness.cs (hunk 1 above)
Comment (mandel-macaque) [reply tests/xharness/Harness.cs]: Ah, perfect!

Comment (mandel-macaque) [reply tests/xharness/Harness.cs]: Updated after the chat we had in slack.

**Comment by mandel-macaque** [comment]
Jenkins is down for maintenance. We might have to retry this later when ever is up.

**Pull request merged**